File: CoffeeMachine.java
import java.util.Scanner;

public class CoffeeMachine {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int[] coffeeMachine = {400, 540, 120, 9, 550};                   // water, milk, coffee, cups, money
        boolean running = true;
        while (running) {
            System.out.println("Write action (buy, fill, take, remaining, exit):");
            switch (sc.next()) {
                case "buy" :
                    buy(coffeeMachine);
                    break;
                case "fill" :
                    fill(coffeeMachine);
                    break;
                case "take" :
                    take(coffeeMachine);
                    break;
                case "remaining" :
                    showSupply(coffeeMachine);
                    break;
                case "exit" :
                    running = false;
                    break;
            }
        }
    }
    public static void showSupply(int arr[]) {
        System.out.println("The coffee machine has:");
        System.out.println(arr[0] + " of water");
        System.out.println(arr[1] + " of milk");
        System.out.println(arr[2] + " of coffee beans");
        System.out.println(arr[3] + " of disposable cups");
        System.out.println(arr[4] + " of money");
    }

    public static void buy(int arr[]) {
        System.out.println("What do you want to buy? 1 - espresso, 2 - latte, 3 - cappuccino, back - to main menu:");
        Scanner scanner = new Scanner(System.in);
        switch (scanner.nextLine()) {
            case "1" :
                if (arr[0] < 250) {
                    System.out.println("Sorry, not enough water!");
                    break;
                }
                if (arr[2] < 16) {
                    System.out.println("Sorry, not enough coffee beans!");
                    break;
                }
                if (arr[3] < 1) {
                    System.out.println("Sorry, not enough cups!");
                    break;
                }
                System.out.println("I have enough resources, making you a coffee!");
                arr[0] -= 250;
                arr[2] -= 16;
                arr[3] -= 1;
                arr[4] += 4;
                break;
            case "2" :
                if (arr[0] < 350) {
                    System.out.println("Sorry, not enough water!");
                    break;
                }
                if (arr[0] < 75) {
                    System.out.println("Sorry, not enough milk!");
                    break;
                }
                if (arr[2] < 20) {
                    System.out.println("Sorry, not enough coffee beans!");
                    break;
                }
                if (arr[3] < 1) {
                    System.out.println("Sorry, not enough cups!");
                    break;
                }
                System.out.println("I have enough resources, making you a coffee!");
                arr[0] -= 350;
                arr[1] -= 75;
                arr[2] -= 20;
                arr[3] -= 1;
                arr[4] += 7;
                break;
            case "3" :
                if (arr[0] < 200) {
                    System.out.println("Sorry, not enough water!");
                    break;
                }
                if (arr[0] < 200) {
                    System.out.println("Sorry, not enough milk!");
                    break;
                }
                if (arr[2] < 12) {
                    System.out.println("Sorry, not enough coffee beans!");
                    break;
                }
                if (arr[3] < 1) {
                    System.out.println("Sorry, not enough cups!");
                    break;
                }
                System.out.println("I have enough resources, making you a coffee!");
                arr[0] -= 200;
                arr[1] -= 100;
                arr[2] -= 12;
                arr[3] -= 1;
                arr[4] += 6;
                break;
            default :
                break;
        }
    }

    public static void fill(int arr[]) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Write how many ml of water do you want to add:");
        arr[0] += scanner.nextInt();
        System.out.println("Write how many ml of milk do you want to add:");
        arr[1] += scanner.nextInt();
        System.out.println("Write how many grams of coffee beans do you want to add:");
        arr[2] += scanner.nextInt();
        System.out.println("Write how many disposable cups of coffee do you want to add:");
        arr[3] += scanner.nextInt();
    }

    public static void take(int arr[]) {
        System.out.println("I gave you $" + arr[4]);
        arr[4] = 0;
    }
}

File: CoffeeMachine.py
class CoffeeMachine:
    water = 0
    milk = 0
    beans = 0
    cups = 0
    money = 0
    action = None
    working = False

    def __init__(self):
        self.water = 400
        self.milk = 540
        self.beans = 120
        self.cups = 9
        self.money = 550
        self.action = "getting action"
        self.working = True

    def get_action(self):
        if self.action == "getting action":
            self.action = input("Write action (buy, fill, take, remaining, exit):\n")
        elif self.action == "buy":
            return input("What do you want to buy? 1 - espresso, 2 - latte, 3 - cappuccino, back - to main menu:\n")
        elif self.action == "getting water":
            return int(input("Write how many ml of water do you want to add:\n"))
        elif self.action == "getting milk":
            return int(input("Write how many ml of milk do you want to add:\n"))
        elif self.action == "getting beans":
            return int(input("Write how many grams of coffee beans do you want to add:\n"))
        elif self.action == "getting cups":
            return int(input("Write how many disposable cups of coffee do you want to add:\n"))

    def print_supplies(self):
        print(f'''
        The coffee machine has:
        {self.water} of water
        {self.milk} of milk
        {self.beans} of coffee beans
        {self.cups} of disposable cups
        {self.money} of money \n''')

    def buy(self, x):
        if x == "1":
            if self.water < 250:
                print("Sorry, not enough water!\n")
            elif self.beans < 16:
                print("Sorry, not enough coffee beans!\n")
            elif self.cups < 1:
                print("Sorry, not enough cups!\n")
            else:
                print("I have enough resources, making you a coffee!\n")
                self.water -= 250
                self.beans -= 16
                self.cups -= 1
                self.money += 4
        elif x == "2":
            if self.water < 350:
                print("Sorry, not enough water!\n")
            elif self.milk < 75:
                print("Sorry, not enough milk!\n")
            elif self.beans < 20:
                print("Sorry, not enough coffee beans!\n")
            elif self.cups < 1:
                print("Sorry, not enough cups!\n")
            else:
                print("I have enough resources, making you a coffee!\n")
                self.water -= 350
                self.milk -= 75
                self.beans -= 20
                self.cups -= 1
                self.money += 7
        elif x == "3":
            if self.water < 200:
                print("Sorry, not enough water!\n")
            elif self.milk < 100:
                print("Sorry, not enough milk!\n")
            elif self.beans < 12:
                print("Sorry, not enough coffee beans!\n")
            elif self.cups < 1:
                print("Sorry, not enough cups!\n")
            else:
                print("I have enough resources, making you a coffee!\n")
                self.water -= 200
                self.milk -= 100
                self.beans -= 12
                self.cups -= 1
                self.money += 6
        else:
            print()

    def fill(self, water, milk, beans, cups):
        self.water += water
        self.milk += milk
        self.beans += beans
        self.cups += cups

    def take(self):
        print(f"I gave you ${self.money}\n")
        self.money = 0

    def work(self):
        while self.working:
            self.get_action()
            if self.action == "buy":
                choice = self.get_action()
                self.buy(choice)
                self.action = "getting action"
            elif self.action == "fill":
                self.action = "getting water"
                water_income = self.get_action()
                self.action = "getting milk"
                milk_income = self.get_action()
                self.action = "getting beans"
                beans_income = self.get_action()
                self.action = "getting cups"
                cups_income = self.get_action()
                self.fill(water_income, milk_income, beans_income, cups_income)
                self.action = "getting action"
            elif self.action == "take":
                self.take()
                self.action = "getting action"
            elif self.action == "remaining":
                self.print_supplies()
                self.action = "getting action"
            elif self.action == "exit":
                self.action = "getting action"
                self.working = False


coffee_machine_1 = CoffeeMachine()
coffee_machine_1.work()
